File: nagasenacs/src/Nagasena/Proc/Grammars/EXIGrammarUse.cs
using System.Diagnostics;

using EventType = Nagasena.Proc.Common.EventType;
using EventTypeList = Nagasena.Proc.Common.EventTypeList;
using IGrammar = Nagasena.Proc.Common.IGrammar;

namespace Nagasena.Proc.Grammars {

  internal sealed class EXIGrammarUse : Grammar, IGrammar {

    internal EXIGrammar exiGrammar;
    internal readonly int contentDatatype;

    internal EXIGrammarUse(int contentDatatype, GrammarCache grammarCache) : base(SCHEMA_GRAMMAR_ELEMENT_AND_TYPE_USE, grammarCache) {
      this.contentDatatype = contentDatatype;
    }

    public override void init(GrammarState stateVariables) {
      exiGrammar.init(stateVariables);
      stateVariables.contentDatatype = contentDatatype;
    }

    public override bool SchemaInformed {
      get {
        Debug.Assert(false);
        return true;
      }
    }

    ///////////////////////////////////////////////////////////////////////////
    /// Implementation of IGrammar (used by StringTable)
    ///////////////////////////////////////////////////////////////////////////

    public void reset() {
      Debug.Assert(false);
    }

    ///////////////////////////////////////////////////////////////////////////
    /// Implementation of abstract methods
    ///////////////////////////////////////////////////////////////////////////

    internal override void attribute(EventType eventType, GrammarState stateVariables) {
      Debug.Assert(false);
    }

    internal override EventTypeList getNextEventTypes(GrammarState stateVariables) {
      Debug.Assert(false);
      return null;
    }

    internal override EventCodeTuple getNextEventCodes(GrammarState stateVariables) {
      Debug.Assert(false);
      return null;
    }

    public override void element(EventType eventType, GrammarState stateVariables) {
      Debug.Assert(false);
    }

    internal override Grammar wildcardElement(int eventTypeIndex, int uriId, int localNameId, GrammarState stateVariables) {
      Debug.Assert(false);
      return null;
    }

    internal override void xsitp(int tp, GrammarState stateVariables) {
      Debug.Assert(false);
    }

    internal override void nillify(int eventTypeIndex, GrammarState stateVariables) {
      Debug.Assert(false);
    }

    public override void chars(EventType eventType, GrammarState stateVariables) {
      Debug.Assert(false);
    }

    public override void undeclaredChars(int eventTypeIndex, GrammarState stateVariables) {
      Debug.Assert(false);
    }

    public override void miscContent(int eventTypeIndex, GrammarState stateVariables) {
      Debug.Assert(false);
    }

    public override void end(GrammarState stateVariables) {
      Debug.Assert(false);
    }

  }

}

File: nagasena/src/org/openexi/proc/grammars/EXIGrammarUse.java
package org.openexi.proc.grammars;

import org.openexi.proc.common.EventType;
import org.openexi.proc.common.EventTypeList;
import org.openexi.proc.common.IGrammar;

final class EXIGrammarUse extends Grammar implements IGrammar {
  
  EXIGrammar exiGrammar;
  final int contentDatatype;

  EXIGrammarUse(int contentDatatype, GrammarCache grammarCache) {
    super(SCHEMA_GRAMMAR_ELEMENT_AND_TYPE_USE, grammarCache);
    this.contentDatatype = contentDatatype;
  }

  @Override
  public void init(GrammarState stateVariables) {
    exiGrammar.init(stateVariables);
    stateVariables.contentDatatype = contentDatatype;
  }

  @Override
  public boolean isSchemaInformed() {
    assert false;
    return true;
  }
  
  ///////////////////////////////////////////////////////////////////////////
  /// Implementation of IGrammar (used by StringTable)
  ///////////////////////////////////////////////////////////////////////////

  public void reset() {
    assert false;
  }

  ///////////////////////////////////////////////////////////////////////////
  /// Implementation of abstract methods
  ///////////////////////////////////////////////////////////////////////////

  @Override
  void attribute(EventType eventType, GrammarState stateVariables) {
    assert false;
  }
  
  @Override
  EventTypeList getNextEventTypes(GrammarState stateVariables) {
    assert false;
    return null;
  }

  @Override
  EventCodeTuple getNextEventCodes(GrammarState stateVariables) {
    assert false;
    return null;
  }

  @Override
  public void element(EventType eventType, GrammarState stateVariables) {
    assert false;
  }

  @Override
  Grammar wildcardElement(int eventTypeIndex, int uriId, int localNameId, GrammarState stateVariables) {
    assert false;
    return null;
  }

  @Override
  void xsitp(int tp, GrammarState stateVariables) {
    assert false;
  }

  @Override
  void nillify(int eventTypeIndex, GrammarState stateVariables) {
    assert false;
  }
  
  @Override
  public void chars(EventType eventType, GrammarState stateVariables) {
    assert false;
  }

  @Override
  public void undeclaredChars(int eventTypeIndex, GrammarState stateVariables) {
    assert false;
  }

  @Override
  public void miscContent(int eventTypeIndex, GrammarState stateVariables) {
    assert false;
  }

  @Override
  public void end(GrammarState stateVariables) {
    assert false;
  }
  
}
